app/services/authorization_service.rb

```diff
@@ -0,0 +1,11 @@
+class AuthorizationService
+  def self.call(subject, category, verb, params, object)
+    rule = Authorization::Rules::PERMISSIONS.dig(subject, category.to_sym, verb.to_sym)
+
+    if rule.respond_to?(:call)
+      rule.call(object, params)
+    else
+      rule
+    end
+  end
+end
```

Comment:
Not sure I love the giant hash....

however the only alternative I have is:
```
class AuthorizationService
  def self.call(subject, category, verb, params, object)
    case [subject, category.to_sym, verb.to_sym]
    when %i[provider subscribers create], %i[provider submissions index], %i[provider submissions show],
        %i[provider submissions create], ...
      true
    when %i[provider subscribers destroy]
      !object || object.subscriber_type == "provider"
    when %i[provider submissions update]
      state_pair_allowed?(object, params, PERMITTED_SUBMISSION_STATE_CHANGES[:provider])
      ...
    end
  end
end
```
which I am not sure is better

Reply:
Yeah, if it gets much bigger maybe we switch to something like https://github.com/CanCanCommunity/cancancan, but that's probably a bit heavyweight for the current requirement. In the meantime, most of our rules are basically just config at the moment, so I was almost tempted to define it in a yaml file instead of a hash. But again, for the number of rules, the setup to parse and apply yaml is a bit heavy at the mo. I think so long as it's easy to read and understand what each type of user is allowed to do, the hash is the least worst for now.